src/raft.c

```diff
@@ -272,20 +270,23 @@ static RRStatus validateRaftRedisCommandArray(RedisRaftCtx *rr, RedisModuleCtx *
  */
 static RRStatus handleSharding(RedisRaftCtx *rr, RedisModuleCtx *ctx, RaftRedisCommandArray *cmds)
 {
+    unsigned int slot;
+
     if (!isSharding(rr)) {
         return RR_OK;
     }
 
-    if (computeHashSlot(rr, ctx, cmds) != RR_OK) {
-        return RR_ERROR;
+    /* If commands have no keys, continue */
+    if (!IsKeyCommands(rr, ctx, cmds)) {
```

Comment:
I think we duplicate the work here. We loop over the command array and pull the keys. We are going to do the same thing in `computeHashSlot()`. I know we did this to avoid cast for `slot` but I'm not sure it is worth the effort. Maybe, we should just go with cast?